docs/ai/how-to/use-redis-for-memory.md

```diff
@@ -0,0 +1,64 @@
+---
+title: "Use Redis for Memory Storage with the Semantic Kernel SDK"
+description: "Learn how to use a Redis database with the RediSearch module to give your AI memories in Semantic Kernel SDK for .NET."
+author: haywoodsloan
+ms.topic: how-to
+ms.date: 04/17/2024
+
+#customer intent: As a .NET developer, I want to use Redis for memory storage with the Semantic Kernel SDK so that I can store and recall memories in my application.
+
+---
+
+# Use Redis for memory storage with the Semantic Kernel SDK
+
+This article demonstrates how to integrate a Redis database with the RediSearch module into the [Semantic Kernel SDK](/semantic-kernel/overview) and use it for memory storage and retrieval.
+
+[Memories](/semantic-kernel/memories/) represent text information that has been stored alongside a precomputed embedding vector for the whole text. When an LLM is prompted to recall a memory, it uses these precomputed embeddings to efficiently evaluate if a memory is relevant to the prompt. Once the LLM finds a matching memory, it uses the memory's text information as context for the next steps in the prompt completion.
+
+When you add memory storage to the Semantic Kernel SDK, it provides a broader context for your requests. It also enables you to store data like a traditional database but query it using natural language.
+
+## Prerequisites
+
+* An Azure account with an active subscription. [Create an account for free](https://azure.microsoft.com/free/?WT.mc_id=A261C142F).
+* [.NET SDK](https://dotnet.microsoft.com/download/visual-studio-sdks)
+* [`Microsoft.SemanticKernel` NuGet package](https://www.nuget.org/packages/Microsoft.SemanticKernel)
+* [`Microsoft.SemanticKernel.Connectors.Redis` NuGet package](https://www.nuget.org/packages/Microsoft.SemanticKernel.Connectors.Redis)
+* [`Microsoft.SemanticKernel.Plugins.Memory` NuGet package](https://www.nuget.org/packages/Microsoft.SemanticKernel.Plugins.Memory)
+* [`StackExchange.Redis` NuGet package](https://www.nuget.org/packages/StackExchange.Redis)
+* A Redis database with the RediSearch module, [deployed and accessible to your .NET application](/azure/azure-cache-for-redis/quickstart-create-redis-enterprise)
```

Comment:
```suggestion
* A Redis database that has the RediSearch module, [deployed and accessible to your .NET application](/azure/azure-cache-for-redis/quickstart-create-redis-enterprise)
```
Additional change: Currently, this sentence indicates that the Redis database is deployed and accessible. Is that what we mean? If we mean that the RediSearch module is deployed and accessible, please replace the comma after "module" with "that's" ("module that's deployed").

Reply:
Is the Redis database that's deployed and accessible it just also needs the RediSearch module enabled.